### PR TITLE
Refactor the SearchFiltersWrapper 

### PR DESCRIPTION
**Pull request by nickgraz**
- Separate the FeaturedFilters and IconFilters into separate components
- I added a note about some improvements that could be made to this wrapper in the future, which should be done if we ever need to use this wrapper with _only_ featured filters, as opposed to featured and icon filters together.
- With this wrapper currently, FeaturedFilters can only be included if there are filtersProps that also include IconFilters

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2131?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #2102** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2102?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #2103** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2103?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #2117** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2117?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #2128** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2128?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #2129** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2129?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #2130** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2130?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #2131** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2131?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2131?utm_source=stack-comment).

File: src/components/common/SearchFiltersWrapper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+// A "featured filter" is one that resides within its own multi select, between the search bar and the filters hidden
+// behind the filter icon (those are now called "icon filters"). Since the implementer should not care about where the
+// filters are (featured or icon), the `filtersProps.setFilters` and `filterProps.filters` passed in are really used to
+// control "all filters". Ideally the SearchFiltersProps gets turned into IconFiltersProps, and those two aforementioned
+// properties get put at the top level of the SearchProps, to make it more obvious. That is a slightly bigger refactor
+// than I want to embark on right now since there are a good number of places using this component. One side effect of
+// this is that **you can't use featured filters without the icon filters also being set up.** As soon as we decide to
+// implement this wrapper with only the featured filters, it makes sense to do the refactor. Hopefully this note
+// clears up any confusion and serves as a reminder that we should do that in the future.
```
Comment (karthikbtf) [inline src/components/common/SearchFiltersWrapper/index.tsx:25]: Illustrate what the props might look like, just to help further understand.

**Review by karthikbtf — APPROVED** [review]
Thanks for this refactor 🙌

File: src/components/common/SearchFiltersWrapper/index.tsx (hunk 1 above)
Comment (nickgraz) [reply src/components/common/SearchFiltersWrapper/index.tsx]: Let me know what you think!

Comment (karthikbtf) [reply src/components/common/SearchFiltersWrapper/index.tsx]: Thanks @nickgraz this is super helpful 🙏 and I get the picture now!!

**Comment by nickgraz** [comment]
### Merge activity

* **Feb 7, 1:21 PM EST**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2131).

**Pull request merged**